packages/react-composites/src/composites/MeetingComposite/ChatButton.tsx

```diff
@@ -4,24 +4,38 @@
 import React from 'react';
 import { ControlBarButton, ControlBarButtonProps } from '@internal/react-components';
 import { Chat20Filled, Chat20Regular } from '@fluentui/react-icons';
+import { NotificationIcon } from './NotificationIcon';
+import { Stack } from '@fluentui/react';
+
+/**
+ * @private
+ */
+export interface ChatButtonProps extends ControlBarButtonProps {
+  unreadMessageCount?: number;
+}
 
 const onRenderOnIcon = (): JSX.Element => <Chat20Filled key={'chatOnIconKey'} primaryFill="currentColor" />;
 const onRenderOffIcon = (): JSX.Element => <Chat20Regular key={'chatOffIconKey'} primaryFill="currentColor" />;
-
 /**
  * @private
  */
-export const ChatButton = (props: ControlBarButtonProps): JSX.Element => {
+export const ChatButton = (props: ChatButtonProps): JSX.Element => {
   const strings = { label: props.label, ...props.strings };
+  const { unreadMessageCount } = props;
 
   return (
-    <ControlBarButton
-      {...props}
-      labelKey={'chatButtonLabelKey'}
-      strings={strings}
-      onRenderOnIcon={props.onRenderOnIcon ?? onRenderOnIcon}
-      onRenderOffIcon={props.onRenderOffIcon ?? onRenderOffIcon}
-      onClick={props.onClick}
-    />
+    <Stack style={{ position: 'relative' }}>
+      <ControlBarButton
+        {...props}
+        labelKey={'chatButtonLabelKey'}
+        strings={strings}
+        onRenderOnIcon={props.onRenderOnIcon ?? onRenderOnIcon}
+        onRenderOffIcon={props.onRenderOffIcon ?? onRenderOffIcon}
+        onClick={props.onClick}
+      />
+      {unreadMessageCount !== undefined && unreadMessageCount > 0 && (
```

Comment:
There is already logic inside `NotificationIcon` that handles different values of `unreadMessageCount`. In this case, I'd argue it's best to keep that logic in a single place: Always return a `<NotificaitonIcon>` and that component decides when to render as `<></>`, including when `unreadMessageCount` is `undefined` or `0`.